auth/src/Repositories/UserRepository.php

```diff
@@ -0,0 +1,61 @@
+<?php namespace Miner\Auth\Repositories;
+
+use Miner\Auth\Entities\User;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class UserRepository
+{
+    public function __construct()
+    {
+        $this->db = app('db');
+    }
+
+    /**
+     * @param string $email
+     * @return User
+     *
+     * @throws \Exception
+     */
+    public function findByEmail(string $email): User
+    {
+        $result = $this->db->select('SELECT id, password_hash, role FROM users WHERE email = ?', [$email]);
```

Comment:
I have been playing around with the idea of moving SQL strings out of models into their own namespaced functions using HEREDOC or better yet, their own `.sql` files that are imported by a class. I have not implemented it yet, but it has been rolling around my brain as a potential pattern.